tools/bench.py

```diff
@@ -0,0 +1,220 @@
+################################################################################
+# Copyright (C) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# SPDX-License-Identifier: Apache-2.0
+################################################################################
+
+import ctypes
+import random
+import timeit
+from typing import List, Sequence, Tuple
+
+import numpy as np
+from gc_mlir import ir, runtime
+from gc_mlir.graph_compiler import GraphCompiler
+from utils import (
+    emit_benchmark_wrapped_main_func,
+    emit_nano_time,
+    get_kernel_func_from_module,
+)
+
+
+def py_timeit_bench(
+    ir_module: ir.Module,
+    entry_name: str,
+    pipeline: str,
+    mlir_args: list,
+    shared_libs: Sequence,
+    ir_printing=False,
+    repeat_time=100,
+    warm_up=20,
+) -> Tuple[float, float]:
+    """benchmark mlir with python timeit."""
+    compiler = GraphCompiler(
+        pipeline,
+        shared_libs,
+    )
+    compile_begin = timeit.default_timer()
+    engine = compiler.compile_and_jit(ir_module, ir_printing=ir_printing)
+    compile_cost = (timeit.default_timer() - compile_begin) * 1000
+
+    func = engine.lookup(entry_name)
+    packed_args = (ctypes.c_void_p * len(mlir_args))()
+    for argNum in range(len(mlir_args)):
```

Comment:
`argNum` -> `arg_num` or `arg_idx`?

Reply:
I copied the code from MLIR[ execution_engine.py](https://github.com/llvm/llvm-project/blob/4283f1ad18db9878b98f98e7a36b4f94ab674d29/mlir/python/mlir/execution_engine.py#L26-L35) so that the cast cost does not affect performance result, and it's more accurate than call the `execution_engine.invoke` directly. Even the naming is not good, I prefer to keep it, and i will add a comment here.